src/main/scala/cats/effect/bio/internals/IORunLoop.scala

```diff
@@ -76,7 +76,7 @@ private[effect] object IORunLoop {
             bRest.push(bFirst)
           }
           bFirst = bindNext.asInstanceOf[Bind[E]]
-          currentIO = fa
+          currentIO = fa.asInstanceOf[Current[E]]
```

Comment:
What is the inferred type here?

Reply:
That was caused by the changed signature of Bind:
`case class Bind[R, E, E1, +A](source: BIO[E, R], f: R => BIO[E1, A])`
The types of errors can change after the `f` is run and it's assumed that there's a `recover` available to handle any potential errors coming from the source BIO. I'm not sure if that's always true (I will have to look into testing that).
I've changed the type of errors to Any in `loop` because that wasn't necessarily true, because of the changed `Bind` signature.

Reply:
👍